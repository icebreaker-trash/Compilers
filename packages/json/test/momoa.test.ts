// @ts-ignore
const { parse } = require('@humanwhocodes/momoa')

describe('momoa', () => {
  it('Common Usage', () => {
    const ast = parse('{"a": 1}', { tokens: true })

    expect(ast).toMatchSnapshot()
  })
})
