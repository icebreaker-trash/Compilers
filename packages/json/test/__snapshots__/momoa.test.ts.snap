// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`momoa Common Usage 1`] = `
{
  "body": {
    "loc": {
      "end": {
        "column": 9,
        "line": 1,
        "offset": 8,
      },
      "start": {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "members": [
      {
        "loc": {
          "end": {
            "column": 8,
            "line": 1,
            "offset": 7,
          },
          "start": {
            "column": 2,
            "line": 1,
            "offset": 1,
          },
        },
        "name": {
          "loc": {
            "end": {
              "column": 5,
              "line": 1,
              "offset": 4,
            },
            "start": {
              "column": 2,
              "line": 1,
              "offset": 1,
            },
          },
          "type": "String",
          "value": "a",
        },
        "type": "Member",
        "value": {
          "loc": {
            "end": {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
            "start": {
              "column": 7,
              "line": 1,
              "offset": 6,
            },
          },
          "type": "Number",
          "value": 1,
        },
      },
    ],
    "type": "Object",
  },
  "loc": {
    "end": {
      "column": 9,
      "line": 1,
      "offset": 8,
    },
    "start": {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "tokens": [
    {
      "loc": {
        "end": {
          "column": 2,
          "line": 1,
          "offset": 1,
        },
        "start": {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "Punctuator",
      "value": "{",
    },
    {
      "loc": {
        "end": {
          "column": 5,
          "line": 1,
          "offset": 4,
        },
        "start": {
          "column": 2,
          "line": 1,
          "offset": 1,
        },
      },
      "type": "String",
      "value": ""a"",
    },
    {
      "loc": {
        "end": {
          "column": 6,
          "line": 1,
          "offset": 5,
        },
        "start": {
          "column": 5,
          "line": 1,
          "offset": 4,
        },
      },
      "type": "Punctuator",
      "value": ":",
    },
    {
      "loc": {
        "end": {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
        "start": {
          "column": 7,
          "line": 1,
          "offset": 6,
        },
      },
      "type": "Number",
      "value": "1",
    },
    {
      "loc": {
        "end": {
          "column": 9,
          "line": 1,
          "offset": 8,
        },
        "start": {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
      },
      "type": "Punctuator",
      "value": "}",
    },
  ],
  "type": "Document",
}
`;
